src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,127 +1,90 @@
+        }
     }
 
-    /**
-     * Верните Optional даты соответствующей дате в массиве.
-     *
-     * @param dateParams Дан массив данных состоящий из 3-х элементов, где:
-     *                   - 1-й элемент массива - год;
-     *                   - 2-й элемент массива - месяц;
-     *                   - 3-й элемент массива - день (число);
-     */
     public Optional<LocalDate> getDate(Integer[] dateParams) {
-        return Optional.empty();
+        try {
+            return Optional.of(LocalDate.of(dateParams[0], dateParams[1], dateParams[2]));
+        } catch (DateTimeException | ArrayIndexOutOfBoundsException exception) {
+            return Optional.empty();
+        }
     }
 
-    /**
-     * Дано время и на сколько часов нужно его изменить.
-     * Верните измененное время на указаную величину.
-     */
     public LocalTime addHours(LocalTime localTime, Integer hoursToAdd) {
-        return LocalTime.now();
+        return localTime.plusHours(hoursToAdd);
     }
 
-    /**
-     * Дано время и на сколько минут нужно его изменить.
-     * Верните измененное время на указаную величину.
-     */
     public LocalTime addMinutes(LocalTime localTime, Integer minutesToAdd) {
-        return LocalTime.now();
+        return localTime.plusMinutes(minutesToAdd);
     }
 
-    /**
-     * Дано время и на сколько секунд нужно его изменить.
-     * Верните измененное время на указаную величину.
-     */
     public LocalTime addSeconds(LocalTime localTime, Integer secondsToAdd) {
-        return LocalTime.now();
+        return localTime.plusSeconds(secondsToAdd);
     }
 
-    /**
-     * Дана дата и на сколько недель нужно ее изменить.
-     * Верните получившуюся дату
-     */
     public LocalDate addWeeks(LocalDate localDate, Integer numberOfWeeks) {
-        return LocalDate.now();
+        return localDate.plusWeeks(numberOfWeeks);
     }
 
-    /**
-     * Дана произвольная дата getDate.
-     * Определите соотношение сегодня к someDate и верните строку:
-     * - "someDate is after текущая дата" - если getDate в будующем
-     * - "someDate is before текущая дата" - если getDate в прошлом
-     * - "someDate is today" - если someDate - сегодня
-     */
     public String beforeOrAfter(LocalDate someDate) {
-        return someDate + "is today";
+        LocalDate currentMoment = LocalDate.now();
+        return someDate.isAfter(currentMoment) ? someDate + " is after " + currentMoment
+                : someDate.isBefore(currentMoment) ? someDate + " is before " + currentMoment
+                : someDate + " is today";
     }
 
-    /**
-     * Дана дата в строковом формате и временная зона.
-     * Верните LocalDateTime в этой временной зоне.
-     * @return LocalDateTime
-     */
     public LocalDateTime getDateInSpecificTimeZone(String dateInString, String zone) {
-        return LocalDateTime.now();
+        return ZonedDateTime.parse(dateInString + "[" + zone + "]").toLocalDateTime();
```

Comment:
let's try to be less straightforward and solve this with the help of Instant class

Reply:
Done